internal/juju/client.go

```diff
@@ -125,13 +126,18 @@ var isJAAS bool
 // IsJAAS uses a synchronisation object to only perform the check once and return the same result.
 func (sc *sharedClient) IsJAAS(defaultVal bool) bool {
 	checkJAASOnce.Do(func() {
+		isJAAS = defaultVal
```

Comment:
issue: I should have caught this earlier. Global variables should be avoided, you can use an `atomic.Bool` instead

Reply:
I've moved the globals into the sharedClient, I think that' the more correct place for them.